### ISSUE TITLE
Request `no-named-as-default` rule options which allows Re-Export `default` syntax

### ISSUE THREAD

**Issue by preco21** [benmosher/eslint-plugin-import#566]
Hello! 👋

Since [`import/no-named-as-default`](https://github.com/benmosher/eslint-plugin-import/blob/master/docs/rules/no-named-as-default.md) rule supports post-ES2015 import syntax, it would be nice if it has some option that allows [Re-Export `default` as `default` syntax](https://github.com/leebyron/ecmascript-export-default-from#exporting-a-default-as-default).

For instance:

`Component/Component.js`:

```javascript
export {
  Component as default,
};
```

`Component/index.js`:

```javascript
export default from './Component'; // not considered problem
```

How about it?

**Comment by benmosher** [comment]
What is the problem with it, currently? Can you provide a failing example? (Could be same code, but with the reported message?)

**Comment by preco21** [comment]
@benmosher Yes. currently, above example is reported:

> Using exported name 'default' as identifier for default export.

**Comment by benmosher** [comment]
That's wild. Definitely sounds like a bug. Will take a look.
